File: Pivot Index/PivotIndex.java
class Solution{
    public int pivotIndex(int[] nums){
        int total = 0;
        for (int num : nums){
            total += num;  // Step 1: Calculate total sum
        }

        int leftSum = 0;
        for (int i = 0; i < nums.length; i++){
            // Step 2: Check if leftSum equals rightSum
            if (leftSum == total - leftSum - nums[i]){
                return i; // Found the pivot index
            }
            leftSum += nums[i];
        }
        return -1;
    }

    public static void main(String[] args){
        Solution sol = new Solution();

        int[] nums = {1,7,3,6,5,6};
        System.out.println(sol.pivotIndex(nums));

        int[] nums2 = {1,2,3};
        System.out.println(sol.pivotIndex(nums2));

        int[] nums3 = {2,1,-1};
        System.out.println(sol.pivotIndex(nums3));
    }
}

File: Pivot Index/pivot_index.py
from typing import List

class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        total = sum(nums)
        left_sum = 0

        for i in range(len(nums)):
            if left_sum == total - left_sum - nums[i]:
                return i
            left_sum += nums[i]

        return -1
    
if __name__ == "__main__":
    sol = Solution()

    nums1 = [1,7,3,6,5,6]
    print("Output 1: ", sol.pivotIndex(nums1))

    nums2 = [1,2,3]
    print("Output 2: ", sol.pivotIndex(nums2))

    nums3 = [2,1,-1]
    print("Output 3: ", sol.pivotIndex(nums3))
